### DISCUSSION TITLE
large scale graphs

### DISCUSSION THREAD

**Discussion by nabsabraham** [stellargraph/stellargraph#1849] [Q&A]
thanks for this awesome project! the integration with neo4j is neat too. Wondering if there's any thoughts on the scale to which I can use stellargraph for my graph problems. ie, can I run any of these algos with networkx graphs that have ~10M nodes, ~15M edges?

**Comment by adocherty** [comment]
Hi, thanks for the question.  How large a graph you can use with StellarGraph depends upon how much memory your graph requires in memory and how much memory your computer has.  The amount of memory taken by a graph depends on a few factors, specifically the number of nodes the graph has and the number of features per node (for feature-based methods) as well as the number of edges in the graph.

We have a page with examples of [how much memory is taken by different example graphs ](https://stellargraph.readthedocs.io/en/stable/demos/zzz-internal-developers/graph-resource-usage.html) that might be useful - note these may be different on your machine.

It could be easiest to try loading your graph with StellarGraph to see if it works!
